uk.co.compucorp.civicrm.civihrsampledata/civihrsampledata.php

```diff
@@ -130,7 +130,9 @@ function _civihrsampledata_importSampleData($extensionDirectory) {
     $fileToImport = new SplFileObject("{$csvDir}/{$file}.csv");
 
     $importerClassName = "CRM_CiviHRSampleData_Importer_{$class}";
-    $importer = new $importerClassName($fileToImport);
+    $importer = new $importerClassName();
+    $importer->setSplFileObject($fileToImport);
```

Comment:
I end up adding a check on the $fileHandler , I just don'k like the idea of  repeat passing the dependencies for all these classes via constructor .